### PR TITLE
[Core][MPI] Improvements to MPI debug utils

### PR DESCRIPTION
**Pull request by roigcarlo**
**📝 Description**
Optimized the python interface of mpi_debug_utils and specializes check functions using type traits.

@KratosMultiphysics/technical-committee the python interface part could be exported to the core to reduce and generalize interfaces if you like.

**🆕 Changelog**
- Using trait specialization in mpi debug utils check
- Generalized python interface of mpi debug utils

### CODE REVIEW

**Comment by roigcarlo** [comment]
Someone?

**Review by loumalouomega — APPROVED** [review]

**Comment by loumalouomega** [comment]
> Someone?

I did some corrections, this way it is faster

**Comment by loumalouomega** [comment]
> > Someone?
> 
> I did some corrections, this way it is faster

I mean that you don't need to modify code and then ask again for approval

File: kratos/mpi/python/add_mpi_debug_utilities_to_python.cpp [hunk 1]
```diff
@@ -18,8 +18,7 @@
 // Project includes
 #include "mpi/utilities/debug_utilities.h"
 
-namespace Kratos {
-namespace Python {
+namespace Kratos::Python {
```
Comment (roigcarlo) [inline kratos/mpi/python/add_mpi_debug_utilities_to_python.cpp:21]: Hey, I did not know that could be used!!!

Comment (rubenzorrilla) [reply kratos/mpi/python/add_mpi_debug_utilities_to_python.cpp:21]: Noob.

**Pull request merged**